src/zope/sendmail/delivery.py

```diff
@@ -103,16 +103,31 @@ def newMessageId(self):
         return "%s@%s" % (left_part, gethostname())
 
     def send(self, fromaddr, toaddrs, message):
-        parser = email.parser.Parser()
-        msg = parser.parsestr(message)
-        messageid = msg.get('Message-Id')
+        # Switch the message to be bytes immediately, any encoding
+        # peculiarities should be handled before.
+        if message is None:
+            header = b''
+        else:
+            if not isinstance(message, bytes):
+                message = message.encode('utf-8')
+            # determine line separator type (assumes consistency)
+            nli = message.find(b'\n')
+            line_sep = b'\n' if nli < 1 or message[nli - 1] != b'\r' \
+                else b'\r\n'
+            header = message.split(line_sep * 2, 1)[0]
```

Comment:
Since we are mostly interested in the first double newline, we could simply use something like

    nlipos = message.find(b'\r\n\r\n')
    if nlipos == -1:
        nlipos = message.find(b'\n\n')
    header = message[:nlipos]

I guess? But I am fine with the current code also.